openedx/core/djangoapps/credit/api.py

```diff
@@ -0,0 +1,102 @@
+""" Contains the APIs for course credit requirements """
+
+from .exceptions import InvalidCreditRequirements
+from .models import CreditCourse, CreditRequirement
+
+
+def set_credit_requirements(course_key, requirements):
+    """ Add requirements to given course
+
+    Args:
+        course_key(CourseKey): The identifier for course
+        requirements(list): List of requirements to be added
+
+    Example:
+        >>> set_credit_requirements(
+                "course-v1-ASUx-DemoX-1T2015",
+                [
+                    {
+                        "namespace": "verification",
+                        "name": "verification",
+                        "criteria": {},
+                    },
+                    {
+                        "namespace": "reverification",
+                        "name": "midterm",
+                        "criteria": {},
+                    },
+                    {
+                        "namespace": "proctored_exam",
+                        "name": "final",
+                        "criteria": {},
+                    },
+                    {
+                        "namespace": "grade",
+                        "name": "grade",
+                        "criteria": {"min_grade": 0.8},
+                    },
+                ])
+
+    Raises:
+        InvalidCreditRequirements
+
+    Returns:
+        None
+    """
+
+    try:
+        credit_course = CreditCourse.get_credit_course(course_key=course_key)
+        for requirement in requirements:
+            CreditRequirement.add_course_requirement(credit_course, requirement)
+    except:  # pylint: disable=bare-except
+        raise InvalidCreditRequirements
```

Comment:
I'd prefer to validate the `requirements` dictionary format at the beginning of this function.  The message of the `InvalidCreditRequirements` exception should describe why the requirements are invalid (for example, "Requirement 2 is missing the 'namespace' field").